### PR TITLE
MULE-11331: Statements leaked when auto key generation fails while ex…

### PR DESCRIPTION
**Pull request by marianogonzalez**
…ecuting stored procedure

### CODE REVIEW

File: modules/db/src/main/java/org/mule/module/db/internal/domain/executor/StoredProcedureExecutor.java [hunk 1]
```diff
@@ -40,7 +40,25 @@ protected Object doExecuteQuery(DbConnection connection, Statement statement, Qu
     @Override
     protected Object doExecuteQuery(DbConnection connection, Statement statement, Query query, AutoGeneratedKeyStrategy autoGeneratedKeyStrategy) throws SQLException
     {
-        autoGeneratedKeyStrategy.execute(statement, query.getQueryTemplate());
+        try
+        {
+            autoGeneratedKeyStrategy.execute(statement, query.getQueryTemplate());
```
Comment (pablokraan) [inline modules/db/src/main/java/org/mule/module/db/internal/domain/executor/StoredProcedureExecutor.java]: This try/catch must go inside the different AutoGeneratedKeyStrategy implementatations (maybe defining an abstract class to avoir repeating the code)

File: modules/db/src/test/java/org/mule/module/db/internal/domain/executor/StoredProcedureExecutorTestCase.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    @Before
+    public void before()
+    {
+        executor = new StoredProcedureExecutor(statementFactory, statementResultHandler);
+    }
+
+    @Test
+    public void statementClosedIfKeyGenerationFails() throws Exception
+    {
+        Statement statement = mock(Statement.class);
```
Comment (pablokraan) [inline modules/db/src/test/java/org/mule/module/db/internal/domain/executor/StoredProcedureExecutorTestCase.java]: There are different methods in the AutoGeneratedKeyStrategy, all of them have the same issue, so a test for each method must be added

File: modules/db/src/main/java/org/mule/module/db/internal/domain/executor/StoredProcedureExecutor.java (hunk 1 above)
Comment (anoordover) [inline modules/db/src/main/java/org/mule/module/db/internal/domain/executor/StoredProcedureExecutor.java]: Is it an idea to implement this using a Decorator/Wrapper pattern around a AutoGeneratedKeyStrategy. You could instantiate this Decorator/Wrapper inside StoredProcedureMessageProcessor.

Comment (pablokraan) [inline modules/db/src/main/java/org/mule/module/db/internal/domain/executor/StoredProcedureExecutor.java]: @anoordover pull requests are welcome

**Pull request merged**